routing/router.go

```diff
@@ -616,373 +450,156 @@ func (r *ChannelRouter) networkHandler() {
-			ChannelID:   channelID,
-			NodeKey1:    msg.FirstNodeID,
-			NodeKey2:    msg.SecondNodeID,
-			BitcoinKey1: msg.FirstBitcoinKey,
-			BitcoinKey2: msg.SecondBitcoinKey,
-			AuthProof: &channeldb.ChannelAuthProof{
-				NodeSig1:    msg.FirstNodeSig,
-				NodeSig2:    msg.SecondNodeSig,
-				BitcoinSig1: msg.FirstBitcoinSig,
-				BitcoinSig2: msg.SecondBitcoinSig,
-			},
-			ChannelPoint: *fundingPoint,
-			// TODO(roasbeef): this is a hack, needs to be removed
-			// after commitment fees are dynamic.
-			Capacity: btcutil.Amount(chanUtxo.Value) - btcutil.Amount(5000),
-		}
-		if err := r.cfg.Graph.AddChannelEdge(edge); err != nil {
-			log.Errorf("unable to add channel: %v", err)
-			return false
+		// TODO(roasbeef): this is a hack, needs to be removed
+		// after commitment fees are dynamic.
+		msg.Capacity = btcutil.Amount(chanUtxo.Value) - btcutil.Amount(5000)
+		msg.ChannelPoint = *fundingPoint
+
+		if err := r.cfg.Graph.AddChannelEdge(msg); err != nil {
+			return errors.Errorf("unable to add msg: %v", err)
 		}
 
 		log.Infof("New channel discovered! Link "+
 			"connects %x and %x with ChannelPoint(%v), chan_id=%v",
-			msg.FirstNodeID.SerializeCompressed(),
-			msg.SecondNodeID.SerializeCompressed(),
-			fundingPoint, channelID)
-
-	// A new authenticated channel update has has arrived, this indicates
-	// that the directional information for an already known channel has
-	// been updated. All updates are signed and validated before reaching
-	// us, so we trust the data to be legitimate.
-	case *lnwire.ChannelUpdateAnnouncement:
-		chanID := msg.ChannelID.ToUint64()
-		edge1Timestamp, edge2Timestamp, _, err := r.cfg.Graph.HasChannelEdge(chanID)
+			msg.NodeKey1.SerializeCompressed(),
+			msg.NodeKey2.SerializeCompressed(),
+			fundingPoint, msg.ChannelID)
+
+	case *channeldb.ChannelEdgePolicy:
+		channelID := lnwire.NewChanIDFromInt(msg.ChannelID)
+		edge1Timestamp, edge2Timestamp, _, err := r.cfg.Graph.HasChannelEdge(msg.ChannelID)
 		if err != nil && err != channeldb.ErrGraphNoEdgesFound {
-			log.Errorf("unable to check for edge existence: %v", err)
-			return false
-		}
+			return errors.Errorf("unable to check for msg "+
+				"existence: %v", err)
 
-		// If the advertised inclusionary block is beyond our knowledge
-		// of the chain tip, then we'll put the announcement in limbo
-		// to be fully verified once we advance forward in the chain.
-		if isPremature(&msg.ChannelID) {
-			blockHeight := msg.ChannelID.BlockHeight
-			log.Infof("Update announcement for chan_id=(%v), is "+
-				"premature: advertises height %v, only height "+
-				"%v is known", chanID, blockHeight,
-				r.bestHeight)
-
-			r.prematureAnnouncements[blockHeight] = append(
-				r.prematureAnnouncements[blockHeight],
-				msg,
-			)
-			return false
 		}
 
-		// As edges are directional edge node has a unique policy for
-		// the direction of the edge they control. Therefore we first
+		// As edges are directional msg msg has a unique policy for
```

Comment:
Looks like a search-and-replace went awry. There're several instances within this diff of `msg` being replaced by `edge`.